pycloudlib/lxd/instance.py

```diff
@@ -195,6 +195,20 @@ def pull_file(self, remote_path, local_path):
         """
         self._log.debug('pulling file %s to %s', remote_path, local_path)
 
+        if self.execute_via_ssh:
+            super().pull_file(remote_path, local_path)
+            return
+
+        if self.series == "xenial":
+            msg = (
+                "Many xenial images do not support pulling files via"
+                " `lxc file pull` due to missing kernel support: you may"
+                " see unavoidable failures.\nSee"
+                " https://github.com/canonical/pycloudlib/issues/132 for"
+                " details."
```

Comment:
As we now have this message three times, I wonder if we should capture the shared text in a separate variable? (Then it would be `.format("lxc file pull")` or something similar.)

Reply:
There is also "pulling files" / "pushing files" / "executing commands", but the warning can probably be made more generic ("the required functionality (format-string) requires lxd-agent ...") and save it in a variable, which can also be used by the tests.

Reply:
Refactored the warning a bit.